src/App.js

```diff
@@ -1,27 +1,69 @@
 import React from 'react';
 import HomePage from '../src/pages/homepage/homepage.component';
-import { Switch , Route} from 'react-router-dom';
+import { Switch , Route, Redirect} from 'react-router-dom';
+import { connect } from 'react-redux';
 import ShopPage from './pages/shop/shop.component';
 import Header from './components/header/header.component';
 import SignInAndSignUpPage from './pages/Sing-In-Sign-Up/Sing-In-Sign-Up.component';
 
 
-
+import {auth, createUserProfileDocument } from './firebase/firebase.utils'; 
 
 import './App.css';
-function App() {
-  return (
+import { setCurrentUser } from './redux/user/user.action';
+
+
+class  App extends React.Component {
+  unsubscribeFromAuth = null;
+
+  componentDidMount() {
+    const { setCurrentUser } =this.props;
+    this.unsubscribeFromAuth = auth.onAuthStateChanged(async userAuth => {
```

Comment:
What's the point of declaring it there ?
This is a class, define it as a method.

Reply:
```js
class App extends React.Component {
  unsubscribeFromAuth = () => {
      const { setCurrentUser } = this.props;

      auth.onAuthStateChanged(async (userAuth) => {
          if (userAuth) {
              const userRef = await createUserProfileDocument(userAuth);

              userRef.onSnapshot((snapShot) => {
                  setCurrentUser({
                      id: snapShot.id,
                      ...snapShot.data(),
                  });
              });
          }
          setCurrentUser(userAuth);
      });

  };

  componentWillUnmount() {
    this.unsubscribeFromAuth();
  }
  render() {}
}
```

Try this.

Reply:
Without componentDidMount() its not working.